pig-tinder-frontend/src/App.js

```diff
@@ -1,37 +1,92 @@
-import React, { useState } from "react";
-import Footer from "./Components/Footer";
-import Header from "./Components/Header";
-import PigEdit from "./Pages/PigEdit";
-import PigIndex from "./Pages/PigIndex";
-import PigNew from "./Pages/PigNew";
-import PigShow from "./Pages/PigShow";
-import Home from "./Pages/Home";
-import NotFound from "./Pages/NotFound";
-import "./App.css";
+  import React, {useState, useEffect} from "react"
+  import Footer from "./Components/Footer";
+  import Header from "./Components/Header";
+  import PigEdit from "./Pages/PigEdit";
+  import PigIndex from "./Pages/PigIndex";
+  import PigNew from "./Pages/PigNew";
+  import PigShow from "./Pages/PigShow";
+  import Home from "./Pages/Home";
+  import NotFound from "./Pages/NotFound";
+  import "./App.css";
+  import mockPigs from "./mockPigs";
+  import { Routes, Route } from "react-router-dom";
+  
+  const App = () => {
+    const [pigs, setPigs] = useState(null);
+    const [newPig, setNewPig] = useState(null);
+    const [updatedPig, setUpdatedPig] = useState(null);
+    const [deletedPig, setDeletedPig] = useState(null);
+  
+    // This function makes an HTTP GET request to read all pigs from your Rails app
+    const fetchPigs = () => {
+      fetch("http://localhost:3000/pigs")
+        .then((response) => response.json())
+        .then((pigs) => setPigs(pigs))
+        .catch((error) => console.error(error));
+    };
+    useEffect(() => {
+      fetchPigs()})
```

Comment:
Does this drop into an infinite loop?

Reply:
Yes it does sir, that ruby terminal kept going